src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,117 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+
+public class MovieRecommender {
+
+    BufferedReader br;
+    BufferedWriter bw;
+    int movies = 0;
+    String currentProduct;
+    String currentScore;
+    String currentUser;
+    HashMap<String, Integer> users;
+    HashMap<String, Integer> products;
+
+    public MovieRecommender(String path) throws IOException {
+        FileInputStream file = null;
+        try {
+            file = new FileInputStream(path);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        GZIPInputStream gzip = new GZIPInputStream(file);
+        Reader reader = new InputStreamReader(gzip);
+        File reviews = new File("dataset.csv");
```

Comment:
Consider substituting hardcoded numbers and strings such as `dataset.csv`, doing this can improve the readability of the program and make easier the process of changing those values.